### PR TITLE
Alms-136 [iOS] Implement batchUpdateDevicePosition in AmazonHttpClient

### PR DESCRIPTION
**Pull request by zeeshanmakeen**
Alms-120 [iOS] Implement getDevicePosition in AmazonHttpClient
Removed Amplify depedency
Fixed test cases

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -13,8 +14,8 @@ let package = Package(
     ],
     dependencies: [
         // Dependencies declare other packages that this package depends on.
-        .package(url: "https://github.com/aws-geospatial/amazon-location-mobile-auth-sdk-ios.git", branch: "main"),
-        .package(url: "https://github.com/aws-amplify/aws-sdk-ios-spm.git", .upToNextMajor(from: "2.0.0"))
+        .package(url: "https://github.com/makeen-project/amazon-location-mobile-auth-sdk-ios.git", branch: "ALMS-134_AmazonHttpClient"),
```
Comment (olegfilimonov) [inline Package.swift]: probably dont wanna commit this to main

File: Package.swift [hunk 2]
```diff
@@ -23,8 +24,8 @@ let package = Package(
             name: "AmazonLocationiOSTrackingSDK",
             dependencies: [
                 .product(name: "AmazonLocationiOSAuthSDK", package: "amazon-location-mobile-auth-sdk-ios"),
-                .product(name: "AWSLocationXCF", package: "aws-sdk-ios-spm"),
-                .product(name: "AWSMobileClientXCF", package: "aws-sdk-ios-spm"),
+//                .product(name: "AWSLocationXCF", package: "aws-sdk-ios-spm"),
```
Comment (olegfilimonov) [inline Package.swift]: no commented out code. just delete it

File: Sources/AWSTrackingSDKiOS/Tracking/Client/LocationTracker.swift [hunk 3]
```diff
@@ -226,40 +229,37 @@ public class LocationTracker {
...
+            return try await sendChunkedLocations(locations: Array(locations.dropFirst()), retries: 3)
+        }
+        else {
+            self.logger.log("Error: \(response.description)")
+            if retries > 0 {
+                self.logger.log("Retrying...")
+                return try await sendChunkedLocations(locations: locations, retries: retries - 1)
+            } else {
+                self.logger.log("Failed after 3 retries")
+                throw NSError() //Error(response.description)
```
Comment (olegfilimonov) [inline Sources/AWSTrackingSDKiOS/Tracking/Client/LocationTracker.swift]: commented out code

**Review by olegfilimonov — CHANGES_REQUESTED** [review]

File: Package.swift (hunk 1 above)
Comment (zeeshanmakeen) [reply Package.swift]: fixed

File: Package.swift (hunk 2 above)
Comment (zeeshanmakeen) [reply Package.swift]: fixed

File: Sources/AWSTrackingSDKiOS/Tracking/Client/LocationTracker.swift (hunk 3 above)
Comment (zeeshanmakeen) [reply Sources/AWSTrackingSDKiOS/Tracking/Client/LocationTracker.swift]: fixed

**Pull request merged**